File: src/common/web/data/entities/features/DataManagementPlanFeature.ts
import { type ProjectFeatureID } from "../EntityTypes";
import { ProjectFeature } from "./ProjectFeature";

/**
 * Data class for the data management plan project feature.
 */
export class DataManagementPlanFeature extends ProjectFeature {
    public static readonly FeatureID: ProjectFeatureID = "dmp";
}

File: src/common/py/data/entities/project/features/data_management_plan_feature.py
import typing
from dataclasses import dataclass, field

from dataclasses_json import dataclass_json

from ...properties.properties_object import PropertyObject
from .project_feature import ProjectFeature, ProjectFeatureID

DataManagementPlan = typing.List[PropertyObject]


@dataclass_json
@dataclass
class DataManagementPlanFeature(ProjectFeature):
    """
    Data class for the data management plan project feature.
    """

    feature_id: typing.ClassVar[ProjectFeatureID] = "dmp"

    plan: DataManagementPlan = field(default_factory=list)
